### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by okoyecharles**
## Description ✍🏽
In this project, we added the [devise](https://github.com/heartcombo/devise) gem to the app and used it for the registration and login of users.

- [x] Remove `current_user` method in ApplicationController, and use `devise` provided.
- [x] Using `devise`
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
   - Ask for confirmation of email.
   - Can reset password.
- [x] [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and my styling.
- [x] Resolve errors

### Participants
* @OkoyeCharles
* @TracK92 

### Comments
Feel free to leave any comments in the PR thread if something is not 100% clear.
### Optional Suggestions
If you notice a problem that is not crucial enough to stop the approval of this PR. However, you strongly recommend I take them into account. Please I would love it if you let me know
### Clarity
Please it would be a pleasure if you could be as clear and precise as possible, as we have only three review attempts.
Thanks for understanding.

---------
_If there are any issues you would like to discuss quickly, you could contact me on [Slack](https://app.slack.com/client/T47CT8XPG/D039GHCKW3F), or (occasionally) on [WhatsApp](https://wa.me/+2348032533432)_

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -41,6 +41,23 @@
...
+    port: 587,
+    domain: 'example.com',
+    user_name: Rails.application.credentials.gmail[:email],
+    password: Rails.application.credentials.gmail[:password],
+    authentication: 'plain',
+    enable_starttls_auto: true
+  }
+
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
+
```
Comment (GzimAsani) [inline config/environments/development.rb]: - [ ] I am getting an error from changes here. Kindly update them
image

File: app/models/User.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (GzimAsani) [inline app/models/User.rb]: - [ ] `:confirmable` needs enabled in the user model (`user.rb` file).

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello    @OkoyeCharles                 :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your login styling it looks great +1: 

 ## Required Changes 

- [x] Check the comments below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/User.rb (hunk 2 above)
Comment (okoyecharles) [reply app/models/User.rb]: @GzimAsani after adding the above suggestion it immediately followed with an error do you have any suggestions on how to solve the problem (changes have been pushed)

**Review by t-yanick — APPROVED** [review]
Hi @OkoyeCharles, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**